narwhals/_pandas_like/series_dt.py

```diff
@@ -270,19 +270,21 @@ def offset_by(self, by: str) -> PandasLikeSeries:
             if unit == "q":
                 multiple *= 3
                 unit = "mo"
-            offset: pd.DateOffset | pd.Timedelta
+            offset: pd.DateOffset | pd.Timedelta | timedelta
```

Comment:
Nit: we *could* omit `pd.Timedelta` from the union, as it is [assignable to](https://typing.python.org/en/latest/spec/concepts.html#the-assignable-to-or-consistent-subtyping-relation) `timedelta` 
```suggestion
            offset: pd.DateOffset | timedelta
```